nilearn/tests/test_helpers.py

```diff
@@ -0,0 +1,127 @@
+        assert str(raised_warning_.message) == expected_warning_
+
+
+def test_next_warn_deprecated_params():
+    mock_input, replacement_params = _mock_args_for_testing_replace_parameter()
+    expected_warnings = [
+        ('The parameter "deprecated_param_0" will be removed in the next Nilearn version. Please use the parameter "replacement_param_0" instead.'),
+        ('The parameter "deprecated_param_1" will be removed in the next Nilearn version. Please use the parameter "replacement_param_1" instead.'),
+        ]
+    with warnings.catch_warnings(record=True) as raised_warnings:
+        helpers._warn_deprecated_params(
+                replacement_params,
+                end_version='next',
+                lib_name='Nilearn',
+                kwargs=mock_input,
+                )
+    for raised_warning_, expected_warning_ in zip(raised_warnings, expected_warnings):
+        assert str(raised_warning_.message) == expected_warning_
+
+
+def test_version_warn_deprecated_params():
+    mock_input, replacement_params = _mock_args_for_testing_replace_parameter()
+    expected_warnings = [
+        ('The parameter "deprecated_param_0" will be removed in Nilearn version 0.6.1rc. Please use the parameter "replacement_param_0" instead.'),
+        ('The parameter "deprecated_param_1" will be removed in Nilearn version 0.6.1rc. Please use the parameter "replacement_param_1" instead.'),
+        ]
+    with warnings.catch_warnings(record=True) as raised_warnings:
+        helpers._warn_deprecated_params(
+                replacement_params,
+                end_version='0.6.1rc',
+                lib_name='Nilearn',
+                kwargs=mock_input,
+                )
+    for raised_warning_, expected_warning_ in zip(raised_warnings,
+                                                  expected_warnings):
+        assert str(raised_warning_.message) == expected_warning_
+
+
+def test_other_lib_warn_deprecated_params():
+    mock_input, replacement_params = _mock_args_for_testing_replace_parameter()
+    expected_warnings = [
+        ('The parameter "deprecated_param_0" will be removed in the next other_lib version. Please use the parameter "replacement_param_0" instead.'),
+        ('The parameter "deprecated_param_1" will be removed in the next other_lib version. Please use the parameter "replacement_param_1" instead.'),
+        ]
+    with warnings.catch_warnings(record=True) as raised_warnings:
+        helpers._warn_deprecated_params(
+                replacement_params,
+                end_version='next',
+                lib_name='other_lib',
+                kwargs=mock_input,
+                )
+    for raised_warning_, expected_warning_ in zip(raised_warnings,
+                                                  expected_warnings):
+        assert str(raised_warning_.message) == expected_warning_
+
+
+def test_replace_parameters():
+    mock_input, replacement_params = _mock_args_for_testing_replace_parameter()
```

Comment:
Could you please add some comments to explain what the tests are probing ? I find them quite abstract/hard to follow.